specs/interop/derivation.md

```diff
@@ -87,18 +98,449 @@ The interop network upgrade timestamp defines the timestamp at which all functio
+- `data`: `0x0x608060405234801561001057600080...` ([full bytecode](../static/bytecode/interop-superchain-e-t-h-bridge-deployment.txt))
+- `sourceHash`: `0x53eccc738e298d613b3c3dcc8ad1d9e9626945a2f7b005252c2b57837176d960`,  
+  computed with the "Upgrade-deposited" type, with `intent = "Interop: SuperchainETHBridge Deployment"`
+
+This results in the Interop SuperchainETHBridge contract being deployed to
+`0x29Abf6CB6036b6dAE44b4B409f836c8Fdc8c6143`, to verify:
+
+```bash
+cast compute-address --nonce=4 0x4210000000000000000000000000000000000004
+Computed Address: 0x29Abf6CB6036b6dAE44b4B409f836c8Fdc8c6143
+```
+
+Verify `sourceHash`:
+
+```bash
+cast keccak $(cast concat-hex 0x0000000000000000000000000000000000000000000000000000000000000002 $(cast keccak "Interop: SuperchainETHBridge Deployment"))
+# 0x53eccc738e298d613b3c3dcc8ad1d9e9626945a2f7b005252c2b57837176d960
+```
+
+Verify `data`:
+
+```bash
+git checkout 71c460ec7c7c05791ddd841b97bcb664a1f0c753
+make build-contracts
+jq -r ".bytecode.object" packages/contracts-bedrock/forge-artifacts/SuperchainETHBridge.sol/SuperchainETHBridge.json
+```
+
+This transaction MUST deploy a contract with the following code hash  
+`0x90a1d78c6e6340d40c0c05a4a30b9bf326280fe95c7690ac964070b477d9fde4`.
+
+To verify the code hash:
+
+```bash
+git checkout 71c460ec7c7c05791ddd841b97bcb664a1f0c753
+make build-contracts
+cast k $(jq -r ".deployedBytecode.object" packages/contracts-bedrock/forge-artifacts/SuperchainETHBridge.sol/SuperchainETHBridge.json)
+```
+
+### SuperchainETHBridge Proxy Update
+
+This transaction updates the SuperchainETHBridge Proxy ERC-1967 implementation
+slot to point to the new SuperchainETHBridge deployment.
+
+A deposit transaction is derived with the following attributes:
+
+- `from`: `0x0000000000000000000000000000000000000000`
+- `to`: `0x4200000000000000000000000000000000000024` (SuperchainETHBridge Proxy)
+- `mint`: `0`
+- `value`: `0`
+- `gasLimit`: `50,000`
+- `data`: `0x3659cfe60000000000000000000000004200000000000000000000000000000000000024`
+- `sourceHash`: `0x50684989256294e3c64949ea1cf5bad586c7e6b91b8b7f21ee9ef7086efe60db`
+  computed with the "Upgrade-deposited" type, with `intent = "Interop: SuperchainETHBridge Proxy Update"`
+
+Verify data:
+
+```bash
+cast concat-hex $(cast sig "upgradeTo(address)") $(cast abi-encode "upgradeTo(address)" 0x29Abf6CB6036b6dAE44b4B409f836c8Fdc8c6143)
+# 0x3659cfe60000000000000000000000004200000000000000000000000000000000000024
+```
+
+Verify `sourceHash`:
+
+```bash
+cast keccak $(cast concat-hex 0x0000000000000000000000000000000000000000000000000000000000000002 $(cast keccak "Interop: SuperchainETHBridge Proxy Update"))
+# 0x50684989256294e3c64949ea1cf5bad586c7e6b91b8b7f21ee9ef7086efe60db
+```
+
+### Mint mint liquidity into ETHLiquidity
+
+This transaction mints liquidity into the ETHLiquidity contract.
+
+A deposit transaction is derived with the following attributes:
+
+- `from`: `0x4200000000000000000000000000000000000024`
```

Comment:
@protolambda am I correct to set the `from` address here to the `SuperchainETHBridge` proxy address, such that `msg.sender != Predeploys.SUPERCHAIN_ETH_BRIDGE` ([source](https://github.com/ethereum-optimism/optimism/blob/46db2fc4e69c2e5e310edfc9b72939900a240e6a/packages/contracts-bedrock/src/L2/ETHLiquidity.sol#L40)?

Reply:
This should be a simple send of value to the contract where the deposit transaction has a `mint` and `value` set to `type(uint248).max`. Realizing now for this to work, the contract needs a `receive` function. Another option would be to deploy a dummy contract that uses self destruct to transfer the value without calling it, this pattern exists inside of `op-faucet`. This would prevent us needing to modify the source code of the `ETHLiquidity` contract